### PR TITLE
[PD-4010] Validate freeform fields of notifications task

### PR DESCRIPTION
**Pull request by tejaswinigutti31**
## What's this PR do?
Validate freeform fields(email/sms) of notifications task for a valid pattern
## Where should the reviewer start?
Check the validation methods in notifications/models.py
## How should this be manually tested?
A couple of unit tests are written to test the functions.
### Separation of Concerns

- [ ] Back end "business logic" is implemented in plain python, Django model classes, and Django forms. (not Django views or other frameworks)
- [ ] Concerns of UI, api, and model are in separate code units.

### Tests

- [ ] Base classes used for python tests don't do too much unnecessary work.

### CODE REVIEW

**Comment by koonsies** [comment]
It'll probably be worthwhile to make sure these validate functions are called in the task's save and/or clean functions.

File: notifications/tests/test_models.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class NotificationTaskModelTest(TestCase):
+    def setUp(self):
+        models.NotificationTask.objects.all().delete()
+
+    def test_validate_freeform_email(self):
+        """
+        Test for invalid email addresses in a given freeform form email address
+        """
+        with self.assertRaises(ValidationError):
+            models.NotificationTask.objects.create(freeform_email='abc@xyz.com, abc.jobs, abc123@xyz.com')
```
Comment (koonsies) [inline notifications/tests/test_models.py]: We should probably also test the positive case. Enter an example that does not raise a validation error.

Comment (tejaswinigutti31) [reply notifications/tests/test_models.py]: Done!

**Comment by koonsies** [comment]
Merge conflicts from the other PR :(

**Pull request merged**